src/Microsoft.Tye.Extensions/Zipkin/ZipkinExtension.cs

```diff
@@ -14,48 +14,73 @@ internal sealed class ZipkinExtension : Extension
     {
         public override Task ProcessAsync(ExtensionContext context, ExtensionConfiguration config)
         {
-            if (context.Operation == ExtensionContext.OperationKind.LocalRun)
+            if (context.Application.Services.Any(s => s.Name == "zipkin"))
             {
-                if (context.Application.Services.Any(s => s.Name == "zipkin"))
-                {
-                    context.Output.WriteDebugLine("zipkin service already configured. Skipping...");
-                }
-                else
+                context.Output.WriteDebugLine("zipkin service already configured. Skipping...");
+            }
+            else
+            {
+                context.Output.WriteDebugLine("Injecting zipkin service...");
+                var service = new ContainerServiceBuilder("zipkin", "openzipkin/zipkin")
                 {
-                    context.Output.WriteDebugLine("Injecting zipkin service...");
-
-                    var service = new ContainerServiceBuilder("zipkin", "openzipkin/zipkin")
+                    Bindings =
                     {
-                        Bindings =
+                        new BindingBuilder()
                         {
-                            new BindingBuilder()
-                            {
-                                Port = 9411,
-                                ContainerPort = 9411,
-                                Protocol = "http",
-                            },
+                            Port = 9411,
+                            ContainerPort = 9411,
+                            Protocol = "http",
                         },
-                    };
-                    context.Application.Services.Add(service);
+                    },
+                };
+                context.Application.Services.Add(service);
 
-                    foreach (var s in context.Application.Services)
+                foreach (var s in context.Application.Services)
+                {
+                    if (object.ReferenceEquals(s, service))
                     {
-                        if (object.ReferenceEquals(s, service))
-                        {
-                            continue;
-                        }
+                        continue;
+                    }
 
-                        // make zipkin available as a dependency of everything.
-                        if (!s.Dependencies.Contains(service.Name))
-                        {
-                            s.Dependencies.Add(service.Name);
-                        }
+                    // make zipkin available as a dependency of everything.
+                    if (!s.Dependencies.Contains(service.Name))
+                    {
+                        s.Dependencies.Add(service.Name);
```

Comment:
@jkotalik - this worked out nicely 👌 - this way the sidecar only has a binding for zipkin.